src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php

```diff
@@ -64,42 +64,41 @@ public function setPriceData($data, $doMapping = true)
      * @return array
      * @throws Exception
      */
-    public function loadCsvData($fileLocation)
+    public function loadCsvData()
     {
         $helper = Mage::helper('ambimax_priceimport');
-        $io = $this->getCsvStream($fileLocation);
+        $fileNames = $helper->getCurrentFile();
 
-        $data = array();
-        $columns = null;
-        $map = array_flip($this->_map);
-        while (false !== ($csvLine = $io->streamReadCsv(',', '""'))) {
+        foreach ($fileNames as $fileName) {
 
-            if (!$columns) {
-                foreach ($csvLine as $field) {
-                    $columns[] = isset($map[$field]) ? $map[$field] : $field;
-                }
-                continue;
-            }
+            $fileLocation = Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_location');
+            $io = $this->getCsvStream($fileLocation, $fileName);
 
-            //build row array
-            $row = array_combine($columns, $csvLine);
-            if (!$sku = $row['sku']) {
-                continue;
-            }
+            $data = array();
+            $columns = null;
+            $map = array_flip($this->_map);
```

Comment:
Is it right, that `$data` will be reset every iteration? You return it with this method. Here only the data of the last iteration would be returned.
And what about multiple entries with the same SKU in different files?
And `$map` does not have to be in the `foreach`, too, in my opinion.